backend/src/main/java/com/engineering/thesis/backend/service/MedicalFacilityService.java

```diff
@@ -0,0 +1,14 @@
+package com.engineering.thesis.backend.service;
+
+import com.engineering.thesis.backend.model.MedicalFacility;
+import com.engineering.thesis.backend.model.Patient;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface MedicalFacilityService {
+    void create(MedicalFacility medicalFacility);
+    void deleteById(Long id);
+    List<MedicalFacility> selectAll();
+    MedicalFacility selectMedicalFacilityById(Long id);
```

Comment:
W typescript mógłbym zrobić coś takiego: 

```void create(<genericType>x);
void deleteById(Long id);
List<genericType> selectAll();
<genericType> selectMedicalFacilityById(Long id);```

to pozwoliłoby na re-używanie tej części kodu z racji że doktor i wizyty itd. wyglądają z puntu widzenia kodu niemal tak samo :O

Reply:
Dalej chodzi o DDD :)

Reply:
Pozostawione bez zmian